Draft against main

ibz-01 opened the pull request.

ibz-01 committed:
ai api integrated

ibz-01 committed:
Added Deletion Feature and Improved Home Page

FatimaSaleem21 commented on src/ai_addtask.js:
```diff
@@ -0,0 +1,152 @@
+
+
+
+const currentTime = new Date()
+
+    
+    const hour = currentTime.getHours();
+    const weekday = currentTime.toLocaleString("en-US", { weekday: "long" });
+    const month = currentTime.toLocaleString("en-US", { month: "long" });
+    const day = currentTime.getDate();
+    const time = currentTime.toLocaleTimeString("en-US", {
+    hour: "numeric",
+    minute: "2-digit",
+    second: "2-digit",
+    hour12: true,
+    });
+
+    const fullDateTime = `${weekday}, ${month} ${day} | ${time}`;
```
This logic can be moved to a function and placed in a helper.js file.

FatimaSaleem21 commented on src/ai_addtask.js:
```diff
@@ -0,0 +1,152 @@
+
+
+
+const SYSTEM_PROMPT = `
```
This can be moved to a constants file

FatimaSaleem21 commented on src/ai_addtask.js:
```diff
@@ -0,0 +1,152 @@
+`;
+
+
+const HF_ACCESS_TOKEN = "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"
```
Same as above

FatimaSaleem21 commented on src/ai_addtask.js:
```diff
@@ -0,0 +1,152 @@
+const hf = new HfInference(HF_ACCESS_TOKEN)
+
+
+export async function getTasks(title,description,deadline,instructions,timeslots)
```
This can be moved to a separate actions or axios files as we are fetching data.

FatimaSaleem21 commented on src/ai_addtask.js:
```diff
@@ -0,0 +1,152 @@
+
+
+        response = await response.json()
+        console.log(response.candidates[0].content.parts[0].text)
```
Console Logs should be removed.

FatimaSaleem21 commented on src/ai_addtask.js:
```diff
@@ -0,0 +1,152 @@
+        response = await response.json()
+        console.log(response.candidates[0].content.parts[0].text)
+
+        let AiOutput = response.candidates[0].content.parts[0].text
```
It might be possible that response is null and candidates[0] can cause a crash

FatimaSaleem21 commented on src/ai_addtask.js:
```diff
@@ -0,0 +1,152 @@
+        } catch (e) {
+        console.error("AI output is not valid JSON", e);
+        return null;
+        }
```
Parsing should be done in a separate function.

FatimaSaleem21 commented on src/ai_addtask.js:
```diff
@@ -0,0 +1,152 @@
+    } catch(err) {
+        console.error(err.message)
+    }
+}
```
EOF expected

FatimaSaleem21 commented on src/components/CarouselData.jsx:
```diff
@@ -0,0 +1,44 @@
+
+    if (dateObj.getTime() === today.getTime()) return "Today";
+    if (dateObj.getTime() === tomorrow.getTime()) return "Tomorrow";
+    return dateString;
```
Create a separate helper function

FatimaSaleem21 reviewed (commented):
We should move some of the code into separate files to make it easier to manage and maintain. Also, let’s clean things up by removing console logs and unnecessary comments. Running a linter would help keep the code consistent.

ibz-01 committed:
Added Utilities Folder and files

Utils folder created and added the following in it:

helper functions
repetitive code
parsing and fetching codes
constants

ibz-01 committed:
Removed comments and console logs

ibz-01 committed:
Improved Collapse Button visibility and design
